os_repository.go

```
@@ -0,0 +1,202 @@
+		req.Header.Set("Authorization", "Bearer "+gc.token)
+
+		err := setCookie(req.Header, gc.host)
+		if err != nil {
+			return nil, fmt.Errorf("Error While Handling Cookie: %s", err)
+		}
+	} else {
+		req.Header.Set("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(gc.username+":"+gc.password)))
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := gc.http
+	httpResp, httpRespError := client.Do(req)
+	if httpRespError != nil {
+		return nil, httpRespError
+	}
+
+	if httpResp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("OS Repositories not found")
+	}
+	var osRepositories []types.OSRepository
+	responseString, _ := extractString(httpResp)
+	err := json.Unmarshal([]byte(responseString), &osRepositories)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing response data for OS repositories: %s", err)
+	}
+	return osRepositories, nil
+}
+
+// GetOSRepositoryByID Gets OS Repository by ID
+func (gc *GatewayClient) GetOSRepositoryByID(id string) (*types.OSRepository, error) {
+	defer TimeSpent("GetOSRepositoryByID", time.Now())
+
+	path := fmt.Sprintf("/api/v1/OSRepository/%v", id)
+	req, httpError := http.NewRequest(http.MethodGet, gc.host+path, nil)
+	if httpError != nil {
+		return nil, httpError
+	}
+
+	if gc.version == "4.0" {
+		req.Header.Set("Authorization", "Bearer "+gc.token)
+
+		err := setCookie(req.Header, gc.host)
+		if err != nil {
+			return nil, fmt.Errorf("Error While Handling Cookie: %s", err)
+		}
+	} else {
+		req.Header.Set("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(gc.username+":"+gc.password)))
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := gc.http
+	httpResp, httpRespError := client.Do(req)
+	if httpRespError != nil {
+		return nil, httpRespError
+	}
+
+	if httpResp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("OS Repository not found")
+	}
+	var osRepository types.OSRepository
+	responseString, _ := extractString(httpResp)
```

err is ignored here while I can see in some places we have check for this, any specific reason to overlook here and in other places?

---

In this case the responseString is used to unmarshall in next line which returns error in case there was error and responseString was empty.